subprojects/internal-integ-testing/src/main/groovy/org/gradle/integtests/fixtures/executer/AbstractGradleExecuter.java

```diff
@@ -1315,14 +1315,20 @@ public GradleExecuter expectDeprecationWarnings(int count) {
     }
 
     @Override
-    public GradleExecuter expectDeprecationWarning(ExpectedDeprecationWarning warning) {
-        expectedDeprecationWarnings.add(warning);
+    public GradleExecuter expectDeprecationWarning(int times, ExpectedDeprecationWarning warning) {
+        if (times <= 0) {
+            throw new IllegalArgumentException("times must be positive");
+        }
+
+        for (int i = 0; i < times; i++) {
+            expectedDeprecationWarnings.add(warning);
+        }
         return this;
     }
 
     @Override
-    public GradleExecuter expectDocumentedDeprecationWarning(ExpectedDeprecationWarning warning) {
-        return expectDeprecationWarning(normalizeDocumentationLink(warning.getMessage()));
+    public GradleExecuter expectDocumentedDeprecationWarning(int times, ExpectedDeprecationWarning warning) {
```

Comment:
❓ I don't understand this. I thought deprecation logger is built in such a way that each deprecation warning only shows up once in the log.

Reply:
I saw that if the deprecation is triggered from different location in the build file, then it counts as a new deprecation.

Reply:
I replaced the new method by using e.g. `5.times { expectDocumentedDeprecationWarning() }`.